src/mmap.rs

```diff
@@ -506,6 +541,50 @@ impl<B: NewBitmap> GuestMemoryMmap<B> {
     }
 }
 
+#[cfg(all(feature = "xen", unix))]
+impl<B: NewBitmap> GuestMemoryMmap<B> {
+    /// Creates a container and allocates anonymous memory for guest memory regions.
+    ///
+    /// Valid memory regions are specified as a slice of (Address, Size) tuples sorted by Address.
+    pub fn from_ranges(ranges: &[(GuestAddress, usize)]) -> result::Result<Self, Error> {
+        // Creates Standard Unix memory mappings as the file isn't available.
+        Self::from_ranges_with_files(ranges.iter().map(|r| {
+            (
+                r.0,
+                r.1,
+                None,
+                crate::mmap_xen::MmapXenFlags::UNIX.bits(),
+                0,
+            )
+        }))
+    }
+
+    /// Creates a container and allocates anonymous memory for guest memory regions.
+    ///
+    /// Valid memory regions are specified as a sequence of
+    /// (Address, Size, Option<FileOffset>, u32, u32) tuples sorted by Address.
```

Comment:
To make the code easier to read and reason about, it would be great to have a structure for these parameters instead.